woocommerce/woocommerce
# Product creation experience: shortcut to add variation price

### All Submissions:






### Changes proposed in this Pull Request:


This PR aims to introduce an easy way for users to add the variation price. It also modifies a few styles to show the action buttons for the variations without hovering over the rows.

Closes #33554.

### How to test the changes in this Pull Request:

1. Go to `Products` > `Add New`.
2. Add a product name and a description.
3. In the `Product data` section select `Variable product`.
4. Go to `Attributes` and create (or select) a product attribute to use for variations.
5. Go to `Variations` and add a few variations (don't modify them).
6. Press `Save changes`.
7. Verify that a warning is shown with the number of variations without price.

![Screen Shot 2022-10-19 at 17 13 28](https://user-images.githubusercontent.com/1314156/196794456-07a33a32-d808-47e8-8dcd-2b76b5a0f107.png)

9. Press `Add price`. A modal like the one below will be visible.
![Screen Shot 2022-10-19 at 16 03 45](https://user-images.githubusercontent.com/1314156/196781258-65f90590-5ba0-416e-b62b-d7ab79ab1478.png)

10. Verify the currency matches the currency used on the site.
11. Add a text (not numbers) and verify that the validation fails. Add a number with multiple decimal points and verify that the error text says: `Please enter a value with one decimal point (%s) without thousand separators`.
12. After adding a valid number the button `OK` should be enabled.
13. Add a valid number and press `Add prices`.
14. Verify that the `Regular price` for every variation is set with the number added in the modal.
15. Verify that the accordion expands after pressing `Edit`.
16. Press `Save changes`. The warning should disappear.
17. Publish the product.
18. Add another variation. Press `Save changes`. The warning should be shown again.
19. Now add another value and verify that the new value has been added to the new variation without modifying the other variations' values.

Now create a new product but select `Variable subscription` instead of `Variable product` in step 3, and continue with the other steps.

### Other information:




### FOR PR REVIEWER ONLY:
